students/Roy_Tate/lesson04/mailroom2.py

```diff
@@ -0,0 +1,184 @@
+        Dear {0},
+
+        We want to thank you for your recent donation of ${1:.2f}. This was
+        very generous.
+
+        Sincerely,
+
+        The Boss
+        '''
+
+# Email template used for generating thank you emails to all donors.
+email_str_annual = '''\n
+        From:    A Charity Thankful For Your Kindness
+        To:      {0}
+        Subject:  This Year's Challenge!
+
+        Dear {0},
+
+        First, we would like to thank you for your continued generosity throughout the 
+        years. Without contributions like yours, the good things that we are able to do 
+        simply would not be possible. 
+
+        Your contributions to date have totaled ${1:.2f}. 
+
+        This years challenge is to see if you can donate more than your current total
+        in one year. This would effectively double your current donations to the 
+        organization!  Remember, it is for a good cause!
+
+        Sincerely,
+
+        The good guys at the best organization'''
+
+
+# Prints the list of keys from donor_dict (in this case, the donor names)
+def print_list():
+    print('{:*^50}'.format('DONOR LIST'))
+    for donor in donor_dict.keys():
+        print(donor)
+
+
+# Helper function to add donations to appropriate donor's account
+def add_donation(name, new_donation):
+    if name in donor_dict.keys():
+        donor_dict[name].append(new_donation)
+    else:
+        donor_dict[name] = [new_donation]
+
+
+# Send formatted email to donor
+def print_email(email_str, donor_name, donation_amount):
+    print(email_str_ty.format(donor_name.title(), float(donation_amount)))
+    print('\n[DONE] Email sent to: ' + donor_name)
+
+
+# Send an email to all donors listing the sum of their donations.
+def email_all():
+    save_path = input('Enter the full path to the directory to save the files.\n'
+                      'Press Enter to save to the current directory: ')
+    print('Generating emails for all donors...')
+    count = 0
+    os.system('mkdir -p ' + save_path)
+    for donor, donations in donor_dict.items():
+        filename = save_path + donor + '.txt'.strip()
+        os.system('touch ' + filename)
+        with open(filename, 'w') as f:
+            f.write(email_str_annual.format(donor.title(), sum(donations)))
+            print('[SUCCESS] Output file -->  ' + str(os.path.abspath(filename)))
+            count += 1
+    print('[DONE] Successfully generated {} emails.'.format(str(count)))
+
+
+def send_thank_you():
+    # Obtain the name of the donor, the contribution amount, add the donation to the list, and generate
+    # a thank you email.
+
+    name = input("Enter the full name of the recipient (Type 'list' to view donors. Type '1' for Main Menu):\n")
+    # thank_you_args = {
+    #                'list': print_list,
+    #                '1': main,
+    # }
+    # thank_you_prompt = "Enter the full name of the recipient (Type 'list' to view donors. " \
+    #                    "Type '1' for Main Menu):\n"
```

Comment:
It's a good idea to remove commented code from your files before committing. Since git stores a history of your entire project (as long as you commit your changes), there's never really a need to keep old dead code lying around